src/libexpr/primops.cc

```diff
@@ -56,6 +56,7 @@ void EvalState::realiseContext(const PathSet & context)
             throw InvalidPathError(store->printStorePath(ctx));
         if (!decoded.second.empty() && ctx.isDerivation()) {
             drvs.push_back(StorePathWithOutputs{ctx.clone(), {decoded.second}});
+            importedDrvs.push_back(StorePathWithOutputs{ctx.clone(), {decoded.second}});
```

Comment:
minor nit: Wait maybe instead of `push_back`ing each only, we can insert the entirety of `drv` at once?

Reply:
originally i had:

```
importedDrvs.insert(importedDrvs.end(), drvs.begin(), drvs.end());
```

but that gave me this:

```
In file included from src/libexpr/primops.cc:1:
In file included from /Users/mbauer/Code/nix/precompiled-headers.h:1:
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/algorithm:1919:21: error: 
      object of type 'nix::StorePathWithOutputs' cannot be assigned because its copy assignment
      operator is implicitly deleted
        *--__result = _VSTD::move(*--__last);
                    ^
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/algorithm:1948:19: note: 
      in instantiation of function template specialization
      'std::__1::__move_backward<nix::StorePathWithOutputs *, nix::StorePathWithOutputs *>'
      requested here
    return _VSTD::__move_backward(__unwrap_iter(__first), __unwrap_iter(__last), __unwrap_...
                  ^
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/vector:1757:12: note: in
      instantiation of function template specialization
      'std::__1::move_backward<nix::StorePathWithOutputs *, nix::StorePathWithOutputs *>'
      requested here
    _VSTD::move_backward(__from_s, __from_s + __n, __old_last);
           ^
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/vector:2019:17: note: in
      instantiation of member function 'std::__1::vector<nix::StorePathWithOutputs,
      std::__1::allocator<nix::StorePathWithOutputs> >::__move_range' requested here
                __move_range(__p, __old_last, __p + __old_n);
                ^
src/libexpr/primops.cc:74:18: note: in instantiation of function template specialization
      'std::__1::vector<nix::StorePathWithOutputs, std::__1::allocator<nix::StorePathWithOutputs>
      >::insert<std::__1::__wrap_iter<nix::StorePathWithOutputs *> >' requested here
    importedDrvs.insert(importedDrvs.end(), drvs.begin(), drvs.end());
                 ^
src/libstore/path.hh:78:15: note: copy assignment operator of 'StorePathWithOutputs' is implicitly
      deleted because field 'path' has a deleted copy assignment operator
    StorePath path;
              ^
src/libstore/path.hh:19:20: note: copy assignment operator of 'StorePath' is implicitly deleted
      because base class 'rust::Value<3 * sizeof(void *) + 24, ffi_StorePath_drop>' has a deleted
      copy assignment operator
struct StorePath : rust::Value<3 * sizeof(void *) + 24, ffi_StorePath_drop>
                   ^
src/libutil/rust-ffi.hh:35:5: note: copy assignment operator is implicitly deleted because
      'Value<48, &nix::ffi_StorePath_drop>' has a user-declared move constructor
    Value(Value && other)
    ^
In file included from src/libexpr/primops.cc:1:
In file included from /Users/mbauer/Code/nix/precompiled-headers.h:1:
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/algorithm:1744:19: error: 
      object of type 'nix::StorePathWithOutputs' cannot be assigned because its copy assignment
      operator is implicitly deleted
        *__result = *__first;
                  ^
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/algorithm:1769:19: note: 
      in instantiation of function template specialization
      'std::__1::__copy<std::__1::__wrap_iter<nix::StorePathWithOutputs *>,
      nix::StorePathWithOutputs *>' requested here
    return _VSTD::__copy(__unwrap_iter(__first), __unwrap_iter(__last), __unwrap_iter(__result));
                  ^
/nix/store/4i88rry2ckc928by1dzg0q9w98c9n5dr-libc++-7.1.0/include/c++/v1/vector:2021:24: note: in
      instantiation of function template specialization
      'std::__1::copy<std::__1::__wrap_iter<nix::StorePathWithOutputs *>,
      nix::StorePathWithOutputs *>' requested here
                _VSTD::copy(__first, __m, __p);
                       ^
src/libexpr/primops.cc:74:18: note: in instantiation of function template specialization
      'std::__1::vector<nix::StorePathWithOutputs, std::__1::allocator<nix::StorePathWithOutputs>
      >::insert<std::__1::__wrap_iter<nix::StorePathWithOutputs *> >' requested here
    importedDrvs.insert(importedDrvs.end(), drvs.begin(), drvs.end());
                 ^
src/libstore/path.hh:78:15: note: copy assignment operator of 'StorePathWithOutputs' is implicitly
      deleted because field 'path' has a deleted copy assignment operator
    StorePath path;
              ^
src/libstore/path.hh:19:20: note: copy assignment operator of 'StorePath' is implicitly deleted
      because base class 'rust::Value<3 * sizeof(void *) + 24, ffi_StorePath_drop>' has a deleted
      copy assignment operator
struct StorePath : rust::Value<3 * sizeof(void *) + 24, ffi_StorePath_drop>
                   ^
src/libutil/rust-ffi.hh:35:5: note: copy assignment operator is implicitly deleted because
      'Value<48, &nix::ffi_StorePath_drop>' has a user-declared move constructor
    Value(Value && other)
    ^
2 errors generated.
make: *** [mk/patterns.mk:3: src/libexpr/primops.o] Error 1
```

Reply:
Ughhhh once again we need #3450